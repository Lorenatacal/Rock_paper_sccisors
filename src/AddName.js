import React, { Component } from "react";

class AddName extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      swowMessage: false
    };
  }

  handleChange = e => {
    this.setState({
      name: e.target.value
    });
  };

  onClick = e => {
    this.setState({
      showMessage: true
    });
    e.preventDefault();
  };

  render() {
    const { showMessage } = this.state;

    return (
      <div>
        <input
          type="text"
          onChange={this.handleChange}
          placeholder="Type your name here"
        />
        <button id="button" type="submit" value="Submit" onClick={this.onClick}>
          Submit
        </button>
        {showMessage ? (
          <p>Wecome to the game {this.state.name}</p>
        ) : (
          <p>Please type your name</p>
        )}
      </div>
    );
  }
}

export default AddName;
