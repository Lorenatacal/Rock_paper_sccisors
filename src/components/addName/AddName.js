import React, { Component } from "react";

class AddName extends Component {
  constructor(props) {
    super(props);
    this.state = {
      name: "",
      showMessage: false,
      showInput: true
    };
  }

  handleChange = e => {
    this.setState({
      name: e.target.value
    });
  };

  onClick = e => {
    this.setState({
      showMessage: true,
      showInput: false
    });
    e.preventDefault();
  };

  render() {
    const { showMessage } = this.state;
    const { showInput } = this.state;

    return (
      <div>
        {showInput ? (
          <div>
            <input
              type="text"
              onChange={this.handleChange}
              placeholder="Type your name here"
            />
            <button
              id="button"
              type="submit"
              value="Submit"
              onClick={this.onClick}
            >
              Submit
            </button>
          </div>
        ) : (
          <div />
        )}
        {showMessage ? <p>Wecome to the game {this.state.name}</p> : <p />}
      </div>
    );
  }
}

export default AddName;
