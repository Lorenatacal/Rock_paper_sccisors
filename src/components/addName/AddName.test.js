import React, { Component } from "react";
import AddName from "./AddName";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() });

test("AddName should render correctly without submit", () => {
  const wrapper = Enzyme.shallow(<AddName />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
