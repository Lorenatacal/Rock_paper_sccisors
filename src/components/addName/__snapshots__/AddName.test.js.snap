// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdName should render correctly on handleChange 1`] = `
<div>
  <div />
  <p>
    Wecome to the game 
    Lorena
  </p>
</div>
`;

exports[`AddName should render correctly onClick 1`] = `
<div>
  <div />
  <p>
    Wecome to the game 
  </p>
</div>
`;

exports[`AddName should render correctly without submit 1`] = `
<div>
  <div>
    <input
      data-name="userInput"
      onChange={[Function]}
      placeholder="Type your name here"
      type="text"
    />
    <button
      data-name="SubmitButton"
      id="button"
      onClick={[Function]}
      type="submit"
      value="Submit"
    >
      Submit
    </button>
  </div>
  <p />
</div>
`;
